aws-datastore/src/main/java/com/amplifyframework/datastore/DataStoreGsonFactory.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.datastore;
+
+import com.amplifyframework.api.graphql.GsonResponseAdapters;
+import com.amplifyframework.core.model.query.predicate.GsonPredicateAdapters;
+import com.amplifyframework.core.model.temporal.GsonTemporalAdapters;
+import com.amplifyframework.core.model.types.GsonJavaTypeAdapters;
+import com.amplifyframework.datastore.appsync.ModelWithMetadataDeserializer;
+import com.amplifyframework.datastore.syncengine.TimeBasedUuidTypeAdapter;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+/**
+ * A utility to create Gson instances for the DataStore plugin.
+ */
+public final class DataStoreGsonFactory {
```

Comment:
I'm a little confused by having two separate factories, `ApiGsonFactory`, and `DataStoreGsonFactory`.  From what I can tell, this is how you are using them: 
 - `ApiGsonFactory`
   - API request serialization
   - API response deserialization
   - DataStore response deserialization - this doesn't feel right!
 - `DataStoreApiFactory`
   - DataStore request serialization
   - SQLite (de)serialization of custom types


The DataStore `AppSyncClient` is using the API category under the hood, and hence is using the `ApiGsonFactory` for  all of the network operations - sync, mutate, subscribe.  


I would suggest either:
1. Just have one GsonFactory with everything
2. Have an `ApiGsonFactory` which is used for all API related (de)serialization in both API & DataStore, and a `CustomTypeGsonFactory`, which is used for (de)serialization of custom types to SQLite

Reply:
I should be able to achieve (1). It is the last change I was toying with, prior to putting up the latest version. Let me go back and see that option to its natural conclusion.

Reply:
Awesome.  Other than that, the PR looks great.  Nice cleanup!